cmd/kubectl-gadget/traceloop.go

```diff
@@ -213,183 +133,160 @@ func runTraceloopStop(cmd *cobra.Command, args []string) error {
 	if err != nil {
-		return fmt.Errorf("failed to get traces: %w", err)
+		return err
 	}
 
-	for node, tm := range tracesPerNode {
-		for _, trace := range tm {
-			if trace.TraceID == args[0] {
-				fmt.Printf("%s", utils.ExecPodSimple(client, node,
-					fmt.Sprintf(`curl --silent --unix-socket /run/traceloop.socket 'http://localhost/dump-by-traceid?traceid=%s' ; echo`, args[0])))
-			}
-		}
+	// Print header
+	switch params.OutputMode {
+	case commonutils.OutputModeJSON:
+		// Nothing to print
+	case commonutils.OutputModeColumns:
+		fallthrough
+	case commonutils.OutputModeCustomColumns:
+		fmt.Println(parser.BuildColumnsHeader())
+	default:
+		return commonutils.WrapInErrOutputModeNotSupported(params.OutputMode)
 	}
 
-	return nil
-}
-
-func runTraceloopPod(cmd *cobra.Command, args []string) error {
-	namespace := args[0]
-	podname := args[1]
-	idx := args[2]
+	var traceID string
 
-	client, err := k8sutil.NewClientsetFromConfigFlags(utils.KubernetesConfigFlags)
-	if err != nil {
-		return commonutils.WrapInErrSetupK8sClient(err)
-	}
+	transformEvent := func(line string) string {
+		var events []traceloopTypes.Event
+		if err := json.Unmarshal([]byte(line), &events); err != nil {
+			fmt.Fprintf(os.Stderr, "Error: %s", commonutils.WrapInErrUnmarshalOutput(err, line))
+			return ""
+		}
 
-	pod, err := client.CoreV1().Pods(namespace).Get(context.TODO(), podname, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to get pod %s: %w", podname, err)
-	}
+		for _, event := range events {
+			baseEvent := event.GetBaseEvent()
+			if baseEvent.Type != eventtypes.NORMAL {
+				commonutils.ManageSpecialEvent(baseEvent, params.Verbose)
+				return ""
+			}
 
-	if pod.Spec.NodeName == "" {
-		return fmt.Errorf("pod %s not scheduled yet", podname)
-	}
+			switch params.OutputMode {
+			case commonutils.OutputModeJSON:
+				b, err := json.Marshal(event)
+				if err != nil {
+					fmt.Fprint(os.Stderr, fmt.Sprint(commonutils.WrapInErrMarshalOutput(err)))
+					return ""
+				}
+
+				fmt.Println(string(b))
+			case commonutils.OutputModeColumns:
+				fallthrough
+			case commonutils.OutputModeCustomColumns:
+				fmt.Println(parser.TransformIntoColumns(&event))
+			}
+		}
 
-	fmt.Printf("%s", utils.ExecPodSimple(client, pod.Spec.NodeName,
-		fmt.Sprintf(`curl --silent --unix-socket /run/traceloop.socket 'http://localhost/dump-pod?namespace=%s&podname=%s&idx=%s' ; echo`,
-			namespace, podname, idx)))
+		// HACK Take a look at gadget.go.
+		utils.DeleteTrace(traceID)
 
-	return nil
-}
+		os.Exit(0)
 
-func runTraceloopClose(cmd *cobra.Command, args []string) error {
-	if len(args) != 1 {
-		return commonutils.WrapInErrMissingArgs("<trace-name>")
+		return ""
 	}
 
-	client, err := k8sutil.NewClientsetFromConfigFlags(utils.KubernetesConfigFlags)
-	if err != nil {
-		return commonutils.WrapInErrSetupK8sClient(err)
-	}
+	for _, trace := range traces {
+		if !strings.Contains(trace.Status.Output, containerID) {
+			continue
```

Comment:
trace.Status.Output should be parsed correctly instead of relying on strings.Contains().